lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -28,9 +28,14 @@ import org.partiql.lang.syntax.SqlParser.ParseType.*
 import org.partiql.lang.syntax.TokenType.*
 import org.partiql.lang.syntax.TokenType.KEYWORD
 import org.partiql.lang.util.*
-import java.time.LocalDate
+import java.lang.Integer.min
+import java.time.*
+import java.time.format.DateTimeFormatter
 import java.time.format.DateTimeParseException
 
+// TODO: (Ques) LocalTime has a field nanosecond and is the precision limit for it. Do we want to use this as the default precision if not specified?
+private const val DEFAULT_PRECISION_FOR_TIME = 9
```

Comment:
This is a good question and I am uncertain I know the correct answer.  May need to bring this up in the team channel.

Reply:
I think also we're going to need to truncate the precision according to the value specified in the precision field--this is what postgres does, i.e.:

```
dlurton=# select time (2) '12:30:23.123456';
    time     
-------------
 12:30:23.12
(1 row)

dlurton=# 

```

Reply:
We will round the value (as Postgres does) taking precision into account but that has been delegated to the evaluation phase. I think we can keep the default precision as 9 (nanoseconds precision) as the `java.time` also has a precision of nano seconds.

Reply:
I'm uncertain what "keep the default precision as 9" means here, since my comment above only applies to `time` literals that includes a precision, thus no default value for precision is in play.